### PR TITLE
attempt to compute 'pretty' x-axis ticks

### PR DESCRIPTION
**Pull request by kevinushey**
This PR improves how the X axis ticks are computed and drawn, and ensures that they update when the range grows (e.g. in the case where the total number of steps for training is unknown)

### CODE REVIEW

File: inst/views/components/metrics_charts.html [hunk 1]
```diff
@@ -201,10 +194,20 @@
...
   this.$adjust_axes = function(chart, metric, data) {
+
+    // adjust x-axis range
     var total_epochs = this.$get_total_epochs(data);
     chart.axis.max({
       x: total_epochs
     });
+
+    // define new x axis tick generator
+    var self = this;
```
Comment (jjallaire) [inline inst/views/components/metrics_charts.html]: We have a `thiz` variable already defined for this case so you should just be able to use that.

File: inst/views/components/metrics_charts.html [hunk 2]
```diff
@@ -254,6 +257,45 @@
     request.send();
   };
 
+  // determine ticks for x-axis
+  this.$get_x_axis_ticks = function(total_epochs) {
+
+      // TODO: should depend on width of chart?
+      var total_ticks = 10;
```
Comment (jjallaire) [inline inst/views/components/metrics_charts.html]: My approach was to provide manual ticks for epochs/steps <= 30 as I felt like no matter the width of the display 30 ticks worked pretty well. So I think you could define a static `total_ticks` of anywhere up to 30 and be okay.

File: inst/views/components/metrics_charts.html (hunk 1 above)
Comment (kevinushey) [reply inst/views/components/metrics_charts.html]: Thanks! I missed that.

File: inst/views/components/metrics_charts.html (hunk 2 above)
Comment (kevinushey) [reply inst/views/components/metrics_charts.html]: I went with 10 here as I felt 30 can get pretty crowded in the Viewer pane (especially if the RStudio window is small, and there are a large number of steps [axis ticks with 3 or 4 digits])

**Comment by jjallaire** [comment]
Okay, just tested this out and it's a huge improvement! 

A couple of nits:

1) I noticed that the final tick doesn't get an axis label:

screen shot 2017-09-28 at 5 43 47 am

2) I found one case where the `0` was drawn to the left of the origin:

screen shot 2017-09-28 at 5 47 32 am

**Comment by kevinushey** [comment]
The issue with the final tick being drawn should be fixed up. I'm not sure if there's a way to tell `c3` to not draw the final tick when it's not actually a multiple of the true tick increment though.

No idea what's going on with `0` being drawn left of the origin, but I suspect it would be independent of this PR?

**Comment by jjallaire** [comment]
Before this PR zero isn't drawn at all:

screen shot 2017-09-28 at 12 37 56 pm

Maybe we can just not draw `0` ?

**Comment by kevinushey** [comment]
Sounds good! FWIW I don't have commit access to `tfruns`, if you add me I can just fix that on master?

**Pull request merged**